### PR TITLE
[#2955] refactor(catalogs): use managedStorage capability

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

- remove property `GRAVITINO_MANAGED_ENTITY`
- add managedStorage capability for Hadoop and Kafka catalog

### Why are the changes needed?

Fix: #2955 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

File: catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java [hunk 1]
```diff
@@ -516,10 +515,7 @@ public PropertiesMetadata filesetPropertiesMetadata() throws UnsupportedOperatio
   public void close() throws IOException {}
 
   private Map<String, String> addManagedFlagToProperties(Map<String, String> properties) {
```
Comment (jerryshao) [inline catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java:517]: Why don't you delete this method directly if it is not required?

Comment (mchades) [reply catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java]: fixed

File: catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogCapability.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import com.datastrato.gravitino.connector.capability.CapabilityResult;
+import java.util.Objects;
+
+public class HadoopCatalogCapability implements Capability {
+  @Override
+  public CapabilityResult managedStorage(Scope scope) {
+    if (Objects.requireNonNull(scope) == Scope.SCHEMA) {
+      return CapabilityResult.SUPPORTED;
+    }
+    return CapabilityResult.unsupported("Hadoop catalog does not support managed storage.");
```
Comment (jerryshao) [inline catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogCapability.java:17]: Can you please change to "Hadoop catalog does not support managed storage for {scope}"?

File: catalogs/catalog-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogCapability.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.datastrato.gravitino.connector.capability.CapabilityResult;
+import java.util.Objects;
+
+public class KafkaCatalogCapability implements Capability {
+  @Override
+  public CapabilityResult managedStorage(Scope scope) {
+    if (Objects.requireNonNull(scope) == Scope.SCHEMA) {
+      return CapabilityResult.SUPPORTED;
+    }
+    return CapabilityResult.unsupported("Kafka catalog does not support managed storage.");
```
Comment (jerryshao) [inline catalogs/catalog-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogCapability.java:17]: Also here.

File: catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogCapability.java (hunk 2 above)
Comment (mchades) [reply catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogCapability.java]: updated

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**